File: src/MainService.Models/Entities/Document.cs
namespace MainService.Models.Entities;
public class Document : BaseEntity
{
    public Document() { }
    public Document(string documentPublicId, string documentUrl) {
        PublicId = documentPublicId;
        Url = documentUrl;
    }

    public string? Url { get; set; }
    public string? PublicId { get; set; }
    public int? Size { get; set; }

    public string? ThumbnailUrl { get; set; }
    public string? ThumbnailPublicId { get; set; }
    public int? ThumbnailSize { get; set; }

    public MedicalLicenseDocument MedicalLicenseDocument { get; set; } = null!;
    public UserMedicalInsuranceCompany UserMedicalInsuranceCompany { get; set; } = null!;
}

File: client/src/app/_models/documents/document.ts

export class Document {
  url: string | null = null;
  publicId: string | null = null;
  size: number | null = null;
  thumbnailUrl: string | null = null;
  thumbnailPublicId: string | null = null;
  thumbnailSize: number | null = null;
  name: string | null = null;
  description: string | null = null;
  id: number | null = null;
  createdAt: Date | null = null;

  constructor(init?: Partial<Document>) {
    Object.assign(this, init);
  }
}
